src/test/groovy/com/palantir/gradle/docker/DockerComposePluginTests.groovy

```diff
@@ -137,4 +137,41 @@ class DockerComposePluginTests extends AbstractPluginTest {
         then:
         buildResult.output.contains("Could not find specified template file")
     }
+
+    def 'docker-compose is executed and fails on invalid file'() {
+        given:
+        file('docker-compose.yml') << "FOO"
+        buildFile << '''
+            plugins {
+                id 'com.palantir.docker-compose'
+            }
+        '''.stripIndent()
+        when:
+        BuildResult buildResult = with('dockerComposeUp', "--stacktrace").buildAndFail()
+        then:
+        buildResult.output.contains("Top level")
+    }
+
+    def 'docker-compose successfully creates docker image'() {
+        given:
+        file('docker-compose.yml') << '''
+            version: "2"
+            services:
+              hello:
+                container_name: "helloworld"
+                image: "python:3.4-alpine"
+                command: touch /test/foobarbaz
+                volumes:
+                  - ./:/test
+        '''.stripIndent()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.docker-compose'
+            }
+        '''.stripIndent()
+        when:
+        with('dockerComposeUp').build()
+        then:
+        file("foobarbaz").exists()
```

Comment:
:+1:

Reply:
Yeah thanks for this - was infinitely better than what I'd done first.